src/main/java/com/codecool/keepcash/Service/Authentication/AuthenticationService.java

```diff
@@ -0,0 +1,127 @@
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class AuthenticationService implements UserDetailsService {
+
+    @Autowired
+    private UserDataRepository userDataRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    private AccountService accountService;
+
+    private CategoryService categoryService;
+
+    private ValidationService validationService;
+
+    public AuthenticationService(AccountService accountService,
+                                 CategoryService categoryService,
+                                 ValidationService validationService) {
+        this.accountService = accountService;
+        this.categoryService = categoryService;
+        this.validationService = validationService;
+    }
+
+    public void register(UserRegistrationDto userRegistrationDto) {
+        try {
+            if (isNewDataValid(userRegistrationDto)) {
+                createNewUser(userRegistrationDto);
+            }
+        } catch (NewUserDataException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Transactional
+    public void createNewUser(UserRegistrationDto userRegistrationDto) {
+        User user = new User(bCryptPasswordEncoder.encode(userRegistrationDto.getPassword()),
+                userRegistrationDto.getUsername());
+
+        userRepository.save(user);
+
+        UserData userData = new UserData(userRegistrationDto.getFirstName(),
+                userRegistrationDto.getLastName(),
+                userRegistrationDto.getEmail(),
+                user);
+
+        List<Category> inbuiltCategories = categoryService.createBuiltinCategories();
+        inbuiltCategories.stream().forEach(category -> userData.getCategories().add(category));
+
+        userData.getAccounts().add(accountService.createBuiltinAccounts());
+        userDataRepository.save(userData);
+    }
+
+    private List<ValidationError> newUserDataValidation(UserRegistrationDto userRegistrationDto) {
+        List<List<ValidationError>> listsOfErrors = new ArrayList<>();
+
+        listsOfErrors.add(validationService.registrationDtoNullValidation(userRegistrationDto).getErrors());
```

Comment:
actually, this method is responsible for calling another service, maybe whole these actions can be moved outside?

Reply:
Do you mean putting 89-91 in separated method?